File: android/src/main/kotlin/com/superwall/superwallkit_flutter/json/TriggerResult+Json.kt
import com.superwall.sdk.models.triggers.TriggerResult
import com.superwall.superwallkit_flutter.bridges.createBridgeId

fun TriggerResult.toJson(): Map<String, Any?> {
    return when (this) {
        is TriggerResult.PlacementNotFound -> mapOf("result" to "placementNotFound")
        is TriggerResult.NoAudienceMatch -> mapOf("result" to "noAudienceMatch")
        is TriggerResult.Paywall -> mapOf("result" to "paywall", "experiment" to mapOf("id" to experiment.id,
            "description" to experiment.toString()))
        is TriggerResult.Holdout -> mapOf("result" to "holdout", "experiment" to mapOf("id" to experiment.id,
            "description" to experiment.toString()))
        is TriggerResult.Error -> mapOf("result" to "error", "error" to error.localizedMessage)
    }
}

File: ios/Classes/Json/TriggerResult+Json.swift
import SuperwallKit

extension TriggerResult {
  func toJson() -> [String: Any] {
    switch self {
    case .placementNotFound:
      return ["result": "placementNotFound"]
    case .noAudienceMatch:
      return ["result": "noAudienceMatch"]
    case .paywall(let experiment):
      return ["result": "paywall", "experimentBridgeId": experiment.createBridgeId()]
    case .holdout(let experiment):
      return ["result": "holdout", "experimentBridgeId": experiment.createBridgeId()]
    case .error(let error):
      return ["result": "error", "error": error.localizedDescription]
    }
  }
}
